### PR TITLE
Add pixel, channel, sample information to CameraReadout, update OpticsDescription

### PR DESCRIPTION
**Pull request by maxnoe**
This adapts our instrument data model a bit to get more in line with what we already know from the DL0 data model and it makes the optics tables in the file more consistent.

The `tel_type` attribute is moved from the `TelescopeDescription` to the `OpticsDescription` as `size_type` to avoid overriding the builin `type`.

A new enum `ReflectorShape` is introduced for OpticsDescription and it is deduced from simtel metadata  files where it is available and was added to the known telescopes in the guessing table.

The optics table now only contains one row per optics description, not one per telescope type:

On this branch:
```
optics_name size_type reflector_shape mirror_area num_mirrors num_mirror_tiles equivalent_focal_length effective_focal_length
ASTRI SST SCHWARZSCHILD_COUDER 14.126235008239746 2 2 2.1500000953674316 2.1519100666046143
MST MST HYBRID 106.2413558959961 1 86 16.0 16.445049285888672
LST LST PARABOLIC 386.7332458496094 1 198 28.0 29.30565071105957
```

On master:

```
description name type mirror_area num_mirrors num_mirror_tiles equivalent_focal_length effective_focal_length
MST_MST_NectarCam MST MST 106.2413558959961 1 86 16.0 16.445049285888672
LST_LST_LSTCam LST LST 386.7332458496094 1 198 28.0 29.30565071105957
MST_MST_FlashCam MST MST 106.2413558959961 1 86 16.0 16.445049285888672
SST_ASTRI_CHEC ASTRI SST 14.126235008239746 2 2 2.1500000953674316 2.1519100666046143
```

We also drop support for older data model version.

### CODE REVIEW

**Review by nbiederbeck — APPROVED** [review]

**Review by kosack — DISMISSED** [review]

File: ctapipe/io/datawriter.py [hunk 1]
```diff
@@ -54,7 +54,13 @@ def _get_tel_index(event, tel_id):
             and true parameters.
           - Telescope Impact Parameters were added.
           - Effective focal length and nominal focal length are both included
-            in the optics description now.
+            in the optics description now. Moved `TelescopeDescription.type`
+            to `OpticsDescription.size_type`. Added `OpticsDescription.reflector_shape`.
+          - n_samples, n_samples_long, n_channels and n_pixels are now part
+            of CameraReadout.
+          - The reference_location (EarthLocation origin of the telescope coordinates)
+            is not included in SubarrayDescription
```
Comment (nbiederbeck) [inline ctapipe/io/datawriter.py:62]: typo: now

**Review by nbiederbeck — CHANGES_REQUESTED** [review]

File: ctapipe/io/datawriter.py (hunk 1 above)
Comment (maxnoe) [reply ctapipe/io/datawriter.py]: fixed

**Review by nbiederbeck — DISMISSED** [review]

**Comment by maxnoe** [comment]
@kosack This is ready for review now. I think it is a big improvement.

**Review by kosack — CHANGES_REQUESTED** [review]
The repr of `TelescopeDescription` fails since self.type is still in there:

```py3
>>> source.subarray.tel[1]
AttributeError: 'TelescopeDescription' object has no attribute 'type'
```

but i notice it's in the slots:
```py3
>>> source.subarray.tel[1].__slots__
('name', 'type', 'optics', 'camera')
```

so clearly the type is not getting filled in the contructor.

**Comment by maxnoe** [comment]
@kosack, fixed and test added.

**Comment by maxnoe** [comment]
@kosack shall I rename `camera.camera_name` to `camera.name`? It would be more consistent with the others and not as tautological.

**Comment by maxnoe** [comment]
@kosack done

**Review by nbiederbeck — APPROVED** [review]

**Review by kosack — APPROVED** [review]

**Pull request merged**